### ISSUE TITLE
Extending type of keys

### ISSUE THREAD

**Issue by biface** [biface/ndt#7]
#### Keys

In classic dictionaries, keys are of different types, not just strings. Currently, the extended functions of the NestedDictionary class manipulate strings, which is restrictive: when an instance of the class is created, the keys are of the same type as the other dictionaries.

_The functions that extend the properties of objects in the NestedDictionary class should be improved to handle all types of keys._

#### Keys and hierarchy

The hierarchy of keys within an instance of the NestedDictionary object class are presented as tuples. But these tuples cannot be used to retrieve or assign values.

In other words: `nd[('first',2,("a", "b"))]` must have the same effect as `nd ['first'][2][("a", "b")]`

**Comment by biface** [comment]
Code has been modified in commit 5714831 in order to manage any type of keys.

**Comment by biface** [comment]
Keys and hierarchy will be seen later since I cannot make distinctions between [1,2] and [(1,2)] :

1. [1,2] should be key "1" and then key "2"
2. [(1,2)] should be key (1,2) as a tuple.
